core/core-impl/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhaseFactory.java

```diff
@@ -191,18 +198,112 @@ private UnionMoveSelectorConfig determineDefaultMoveSelectorConfig(HeuristicConf
                 .filter(variableDescriptor -> !variableDescriptor.isListVariable())
                 .distinct()
                 .toList();
+        var hasChainedVariable = basicVariableDescriptorList.stream()
+                .filter(v -> v instanceof BasicVariableDescriptor<Solution_>)
+                .anyMatch(v -> ((BasicVariableDescriptor<?>) v).isChained());
         var listVariableDescriptor = solutionDescriptor.getListVariableDescriptor();
         if (basicVariableDescriptorList.isEmpty()) { // We only have the one list variable.
-            return new UnionMoveSelectorConfig()
-                    .withMoveSelectors(new ListChangeMoveSelectorConfig(), new ListSwapMoveSelectorConfig());
+            if (configPolicy.getNearbyDistanceMeterClass() == null) {
+                return new UnionMoveSelectorConfig()
+                        .withMoveSelectors(new ListChangeMoveSelectorConfig(), new ListSwapMoveSelectorConfig());
+            } else {
+                return new UnionMoveSelectorConfig()
+                        .withMoveSelectors(new ListChangeMoveSelectorConfig(),
+                                new ListSwapMoveSelectorConfig(),
+                                new ListChangeMoveSelectorConfig()
+                                        .withValueSelectorConfig(new ValueSelectorConfig()
+                                                .withId("changeMoveSelector"))
```

Comment:
I'm thinking... I don't remember the specific behavior, but aren't these supposed to be unique? Maybe we want to generate the selector names with some randomness? `changeMoveSelector-f4edab` or something like that.

Reply:
The selectors are created for an empty configuration, and their uniqueness is guaranteed. Anyway, let me introduce some randomness to it.